src/commonMain/kotlin/app/cash/turbine/coroutines.kt

```diff
@@ -15,6 +15,22 @@
  */
 package app.cash.turbine
 
+import kotlin.coroutines.CoroutineContext
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.currentCoroutineContext
+import kotlinx.coroutines.withContext
+
+public const val DEFAULT_TIMEOUT_MS: Long = 1000
+
+/**
+ * Sets the timeout for all [Turbine] instances within this context.
+ */
+public suspend fun <T> withTurbineTimeout(timeoutMs: Long, block: suspend CoroutineScope.() -> T): T {
```

Comment:
Does this need to be public API? When would you not be in control of the turbine instantiation such that you could just pass the desired value as a parameter directly.

Reply:
This kind of scenario is what I was thinking of:

```kotlin

@Test fun myTest() = runTest {
  val subject = Subject(this /* CoroutineScope */, fakeCollaborator)

  subject.doSomething()

  assertThat(fakeCollaborator.outputTurbine.awaitItem()).isEqualTo("hi there")
}
```

Wrapping a block like this with `withTurbineTimeout` makes it possible to assign delays to any fake interactions, even if `.test { ... }` isn't being used.